File: src/main/java/org/shark/example/service/file/SaveFileApiService.java
package org.shark.example.service.file;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.*;

@Slf4j
@Service
public class SaveFileApiService {

    public void start(MultipartFile file) {
        log.info("file name is {}", file.getOriginalFilename());

        try(OutputStream outputStream = new FileOutputStream("file/" + file.getOriginalFilename())) {
            outputStream.write(file.getBytes());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/org/shark/example/service/file/SaveFileApiService.kt
package org.shark.example.service.file

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.web.multipart.MultipartFile
import java.io.File
import java.io.FileOutputStream
import java.io.IOException

@Service
class SaveFileApiService {
    private val log = LoggerFactory.getLogger(SaveFileApiService::class.java)

    fun start(file: MultipartFile) {
        log.info("file name is {}", file.originalFilename)

        try {
            val targetFile = File("file/" + file.originalFilename)
            val parent = targetFile.parentFile
            if (parent != null && !parent.exists()) {
                parent.mkdirs()
            }
            
            FileOutputStream(targetFile).use { outputStream ->
                outputStream.write(file.bytes)
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}
